docs/api.md

```diff
@@ -118,8 +120,93 @@ type AddonsSpec struct {
 }
 
 // TODO maybe we have profiles for scaling behaviors
-type AutoScalerAddonSpec struct {
-    ResourceLimits LimitRange
+type ClusterAutoscaling struct {
+	// Constraints of autoscaling resources
+	ResourceLimits *ResourceLimits `json:"resourceLimits,omitempty"`
+
+	// Configuration of scale down operation
+	ScaleDown *ScaleDownConfig `json:"scaleDown,omitempty"`
+
+	// Gives pods graceful termination time before scaling down
+	MaxPodGracePeriod *int32 `json:"maxPodGracePeriod,omitempty"`
+
+	// Maximum time CA waits for node to be provisioned
+	// +kubebuilder:validation:Pattern=^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
+	MaxNodeProvisionTime string `json:"maxNodeProvisionTime,omitempty"`
+
+	// To allow users to schedule "best-effort" pods, which shouldn't trigger
+	// Cluster Autoscaler actions, but only run when there are spare resources available,
+	// More info: https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#how-does-cluster-autoscaler-work-with-pod-priority-and-preemption
+	PodPriorityThreshold *int32 `json:"podPriorityThreshold,omitempty"`
+
+	// BalanceSimilarNodeGroups enables/disables the
+	// `--balance-similar-node-groups` cluster-autocaler feature.
+	// This feature will automatically identify node groups with
+	// the same instance type and the same set of labels and try
+	// to keep the respective sizes of those node groups balanced.
+	BalanceSimilarNodeGroups *bool `json:"balanceSimilarNodeGroups,omitempty"`
+
+	// Enables/Disables `--ignore-daemonsets-utilization` CA feature flag. Should CA ignore DaemonSet pods when calculating resource utilization for scaling down. false by default
+	IgnoreDaemonsetsUtilization *bool `json:"ignoreDaemonsetsUtilization,omitempty"`
+
+	// Enables/Disables `--skip-nodes-with-local-storage` CA feature flag. If true cluster autoscaler will never delete nodes with pods with local storage, e.g. EmptyDir or HostPath. true by default at autoscaler
+	SkipNodesWithLocalStorage *bool `json:"skipNodesWithLocalStorage,omitempty"`
+}
+
+type ResourceLimits struct {
+	// Maximum number of nodes in all node groups.
+	// Cluster autoscaler will not grow the cluster beyond this number.
+	// +kubebuilder:validation:Minimum=0
+	MaxNodesTotal *int32 `json:"maxNodesTotal,omitempty"`
```

Comment:
What's the default?

Reply:
in the update i've changed this to force the user to specify. i'm not sure what the best option for default max size is, but it seemed like the most important piece of the autoscaling to force a user into specifying.